### PR TITLE
Move K8S logic to providers

### PR DESCRIPTION
**Pull request by otherpirate**

### CODE REVIEW

File: dbaas/workflow/steps/util/host_provider.py [hunk 1]
```diff
@@ -198,6 +198,47 @@ def create_host(self, infra, offering, name, team_name, zone=None,
         host.save()
         return host
 
+    def prepare(self):
```
Comment (elesbom) [inline dbaas/workflow/steps/util/host_provider.py:201]: what prepare means ?

File: dbaas/workflow/steps/util/host_provider.py [hunk 2]
```diff
@@ -217,11 +267,13 @@ def list_zones(self):
...
         return data['zones']
 
-    def host_info(self, host):
+    def host_info(self, host, refresh=False):
         url = "{}/{}/{}/host/{}".format(
             self.credential.endpoint, self.provider, self.environment,
             host.identifier
         )
+        if refresh:
+            url = "{}/refresh".format(url)
```
Comment (elesbom) [inline dbaas/workflow/steps/util/host_provider.py:276]: use urljoin is more safe :D

File: dbaas/workflow/steps/util/k8s.py [hunk 3]
```diff
@@ -165,34 +164,53 @@ def undo(self):
...
 class NewServiceK8S(BaseK8SStep):
+
     def __unicode__(self):
         return "Creating Service on kubernetes..."
 
-    @property
-    def template_path(self):
-        return 'physical/scripts/k8s/service.yaml'
+    def do(self):
+        provider = HostProvider(self.instance, self.environment)
```
Comment (elesbom) [inline dbaas/workflow/steps/util/k8s.py:172]: put that o a property ?

File: dbaas/workflow/steps/util/k8s.py [hunk 4]
```diff
@@ -165,34 +164,53 @@ def undo(self):
...
+        provider.prepare()
+
+    def undo(self):
+        provider = HostProvider(self.instance, self.environment)
```
Comment (elesbom) [inline dbaas/workflow/steps/util/k8s.py:176]: put that property on base ?

File: dbaas/workflow/steps/util/k8s.py [hunk 5]
```diff
@@ -165,34 +164,53 @@ def undo(self):
...
+        provider.remove_configuration()
+
+
+class UpdateHostMetadata(BaseK8SStep):
+    def __unicode__(self):
+        return "Update address of instance and host with pod ip..."
+
+    def do(self):
+        provider = HostProvider(self.instance, self.environment)
```
Comment (elesbom) [inline dbaas/workflow/steps/util/k8s.py:207]: repaeted

File: dbaas/workflow/steps/util/k8s.py (hunk 3 above)
Comment (otherpirate) [reply dbaas/workflow/steps/util/k8s.py:172]: Next PR has a lot of improvement in it

File: dbaas/workflow/steps/util/k8s.py (hunk 4 above)
Comment (otherpirate) [reply dbaas/workflow/steps/util/k8s.py:176]: Same as above

File: dbaas/workflow/steps/util/k8s.py (hunk 5 above)
Comment (otherpirate) [reply dbaas/workflow/steps/util/k8s.py:207]: Same as above

File: dbaas/workflow/steps/util/host_provider.py (hunk 1 above)
Comment (otherpirate) [reply dbaas/workflow/steps/util/host_provider.py:201]: Prepare means that is a step before starting the creation of the host

**Review by elesbom — APPROVED** [review]

**Pull request merged**